compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/collectors/components/DeclarationCheckersDiagnosticComponent.kt

```diff
@@ -39,7 +39,7 @@ class DeclarationCheckersDiagnosticComponent(
     }
 
     override fun visitTypeAlias(typeAlias: FirTypeAlias, data: CheckerContext) {
-        checkers.memberDeclarationCheckers.check(typeAlias, data, reporter)
+        checkers.basicDeclarationCheckers.check(typeAlias, data, reporter)
```

Comment:
Why not both? `(checkers.basicDeclarationCheckers + checkers.memberDeclarationCheckers)` The call to `FirExposedVisibilityDeclarationChecker.checkTypeAlias(declaration, reporter, context)` inside `FirTypeAliasChecker` should not be needed.

You could also make `FirTypeAliasChecker` a `FirMemberDeclarationChecker`... `FirTypeAlias` is a `FirMemberDeclaration`.

Reply:
I think there is no need to run those checkers since a type alias must be top level. Any additional erros found by member checkers seem to be redundant.

Reply:
But, still, `FirTypeAlias` is a subtype of `FirMemberDeclaration` (since it's a member of a file). Thus, you can make `FirTypeAliasChecker` a `FirMemberDeclarationChecker` and use `memberDeclarationCheckers` as before.

Reply:
Ah my bad! I missed Mark's original comment about FirTypeAlias being a member declaration. OK, let me make this a member checker instead.